[docs] [ENG-11196] Update docs for custom builds

radoslawkrzemien opened the pull request:
# Why

https://linear.app/expo/issue/ENG-11196/create-custom-build-example-slacking-team-members
Depends on: https://github.com/expo/eas-build/pull/346

# How

Updated the docs for custom builds, by adding a reference to a new built-in function `eas/send_slack_message` which allows for sending Slack messages from a custom build workflow

# Test Plan

Run automated tests - all pass
Run prettier, lint and lint-prose - no errors
Tested manually
![Screenshot 2024-02-23 at 15 44 27](https://github.com/expo/expo/assets/2974455/53192d72-a71e-473d-a82a-e5e0f6d5a522)
![Screenshot 2024-02-23 at 15 44 40](https://github.com/expo/expo/assets/2974455/2fa7174e-fab7-4045-8cbc-7940f6a10ae9)

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

radoslawkrzemien committed:
[docs] Update docs for custom builds

Added documentation regarding newly added builtin function to send Slack messages from custom builds

See: https://linear.app/expo/issue/ENG-11196/create-custom-build-example-slacking-team-members

szdziedzic commented on docs/pages/custom-builds/schema.mdx:
```diff
@@ -1392,6 +1392,97 @@ build:
+#### `eas/send_slack_message`
+
+Sends a specified message to a configured Slack webhook URL, which then posts it in the related Slack channel.
+For more information on how to use Slack webhooks go to [Sending messages using incoming webhooks](https://api.slack.com/messaging/webhooks).
```
```suggestion
For more information on how to use Slack webhooks check [Slack documentation](https://api.slack.com/messaging/webhooks).
```
What do you think?

szdziedzic commented on docs/pages/custom-builds/schema.mdx:
```diff
@@ -1392,6 +1392,97 @@ build:
+
+| Input               | Description                                                                                                                                                                                                                                                                                                                                                                             ...
+|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `message`           | **string** The text of the message you want to be sent. Example: `'This is the content of the message'`<br />**Note:** You can reference build job properties and other steps outputs in the message for dynamic evaluation. Examples: `'Build URL: ${ eas.job.expoBuildUrl }'`, `Build finished with status: ${ steps.run_fastlane.status_text }`, `Build failed with error: ${ steps. ...
```
```suggestion
| `message`           | **string** The text of the message you want to be sent. Example: `'This is the content of the message'`<br />**Note:** You can reference build job properties and other steps outputs in the message for dynamic evaluation. Examples: `'Build URL: ${ eas.job.expoBuildUrl }'`, `Build finished with status: ${ steps.run_fastlane.status_text }`, `Build failed with error: ${ steps.run_gradle.error_text }`. For more information see [Use output from one step to another](/custom-builds/schema/#use-output-from-one-step-to-another)                                             |
```

szdziedzic reviewed (commented):
Looks reasonable, although the most important review here is from code owners

radoslawkrzemien committed:
[docs] Rephrase

Changed a sentence after review

See: https://linear.app/expo/issue/ENG-11196/create-custom-build-example-slacking-team-members

radoslawkrzemien committed:
[docs] Update link

Changed the URL to a relative one

See: https://linear.app/expo/issue/ENG-11196/create-custom-build-example-slacking-team-members

Simek commented on docs/pages/custom-builds/schema.mdx:
```diff
@@ -1392,6 +1392,97 @@ build:
+build:
+  name: Slack your team from custom build
+  steps:
+    # @info #
+    - eas/send_slack_message:
+        name: Send Slack message to a given webhook URL
+        inputs:
+          message: 'This is a message to plain input URL'
+          slack_hook_url: 'https://hooks.slack.com/services/[rest_of_hook_url]'
+    # @end #
+    # @info #
```
There is not that much value in highlighting the whole actions code, especially without a descriptions.

Let's remove the highlight fully or retain the highlight on important action parts, preferably with tooltips for docs readers.

radoslawkrzemien replied:
I'll leave it one to indicate which is the described function and remove the rest

Simek reviewed (approved):
LGTM, but left a nit about code highlight.

radoslawkrzemien committed:
[docs] Remove most highlights

Kept one highlight and removed the rest

See: https://linear.app/expo/issue/ENG-11196/create-custom-build-example-slacking-team-members

amandeepmittal commented on docs/pages/custom-builds/schema.mdx:
```diff
@@ -1392,6 +1392,83 @@ build:
 
+#### `eas/send_slack_message`
+
+Sends a specified message to a configured Slack webhook URL, which then posts it in the related Slack channel.
+For more information on how to use Slack webhooks check [Slack documentation](https://api.slack.com/messaging/webhooks).
```
```suggestion
Sends a specified message to a configured [Slack webhook URL](https://api.slack.com/messaging/webhooks), which then posts it in the related Slack channel.
```

No need to explicitly call out slack's docs here. We can just link it to the keyword in the first sentence.

amandeepmittal commented on docs/pages/custom-builds/schema.mdx:
```diff
@@ -1392,6 +1392,83 @@ build:
+
+| Input               | Description                                                                                                                                                                                                                                                                                                                                                                             ...
+|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `message`           | **string** The text of the message you want to be sent. Example: `'This is the content of the message'`<br />**Note:** You can reference build job properties and other steps outputs in the message for dynamic evaluation. Examples: `'Build URL: ${ eas.job.expoBuildUrl }'`, `Build finished with status: ${ steps.run_fastlane.status_text }`, `Build failed with error: ${ steps. ...
```
```suggestion
| `message` | **string** The text of the message you want to send. For example, `'This is the content of the message'`.<br /> <br />**Note:** You can reference build job properties and [use other steps outputs](#use-output-from-one-step-to-another) in the message for dynamic evaluation. For example, `'Build URL: ${ eas.job.expoBuildUrl }'`, `Build finished with status: ${ steps.run_fastlane.status_text }`, `Build failed with error: ${ steps.run_gradle.error_text }`. |
```

- Let's break down the note properly from the definition of the property
- Also, highlight the link instead of adding context for using output from one step to another. Use the page's internal link here since the table also exists on the same page as output from one step to another section
- Update verbiage for consistency.

amandeepmittal commented on docs/pages/custom-builds/schema.mdx:
```diff
@@ -1392,6 +1392,83 @@ build:
+| Input               | Description                                                                                                                                                                                                                                                                                                                                                                             ...
+|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ...
+| `message`           | **string** The text of the message you want to be sent. Example: `'This is the content of the message'`<br />**Note:** You can reference build job properties and other steps outputs in the message for dynamic evaluation. Examples: `'Build URL: ${ eas.job.expoBuildUrl }'`, `Build finished with status: ${ steps.run_fastlane.status_text }`, `Build failed with error: ${ steps. ...
+| `slack_hook_url`    | **string** The URL of the previously configured Slack webhook URL, which will post your message to the specified channel. You can provide the plain URL like `slack_hook_url: 'https://hooks.slack.com/services/[rest_of_hook_url]'`, use EAS secrets like `slack_hook_url: ${ eas.env.ANOTHER_SLACK_HOOK_URL }`, or set `SLACK_HOOK_URL` secret which will serve as a default webhook  ...
```
```suggestion
| `slack_hook_url` | **string** The URL of the previously configured Slack webhook URL, which will post your message to the specified channel. You can provide the plain URL like `slack_hook_url: 'https://hooks.slack.com/services/[rest_of_hook_url]'`, use EAS secrets like `slack_hook_url: ${ eas.env.ANOTHER_SLACK_HOOK_URL }`, or set the `SLACK_HOOK_URL` secret, which will serve as a default webhook URL (in this last case, there is no need to provide `slack_hook_url` input). |
```

amandeepmittal reviewed (approved):
Left a few suggestions on how we should format the content and fix the internal link.

radoslawkrzemien committed:
Update docs/pages/custom-builds/schema.mdx

Remove extra sentence and link the keyword

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/custom-builds/schema.mdx

Review tweaks - updated verbiage, removed redundant sentence, linked the keyword

Co-authored-by: Aman Mittal <amandeepmittal@live.com>

radoslawkrzemien committed:
Update docs/pages/custom-builds/schema.mdx

Review tweaks - updated verbiage

Co-authored-by: Aman Mittal <amandeepmittal@live.com>
